source/common/http/conn_manager_impl.cc

```diff
@@ -543,6 +543,9 @@ void ConnectionManagerImpl::ActiveStream::decodeHeaders(HeaderMapPtr&& headers,
 
     if (tracing_decision.is_tracing) {
       active_span_ = connection_manager_.tracer_.startSpan(*this, *request_headers_, request_info_);
+      if (cached_route_.value() && cached_route_.value()->decorator()) {
+        cached_route_.value()->decorator()->apply(*active_span_);
```

Comment:
I think you will need a conn_manager_impl_test to get test coverage for this line.